backend/spec/controllers/spree/admin/orders/customer_details_controller_spec.rb

```diff
@@ -20,23 +20,175 @@
       allow(Spree::Order).to receive_message_chain(:friendly, :find).and_return(order)
     end
 
-    context "#update" do
-      it "does refresh the shipment rates with all shipping methods" do
-        allow(order).to receive_messages(update_attributes: true)
-        allow(order).to receive_messages(next: false)
-        allow(order).to receive_messages(complete?: true)
-        expect(order).to receive(:refresh_shipment_rates)
-          .with(Spree::ShippingMethod::DISPLAY_ON_FRONT_AND_BACK_END)
-        attributes = {
+    describe '#update' do
+      let(:attributes) {
+        {
           order_id: order.number,
           order: {
             email: "",
             use_billing: "",
             bill_address_attributes: {},
             ship_address_attributes: {}
-          }
+          },
+          guest_checkout: 'true'
         }
-        spree_put :update, attributes
+      }
+
+      def send_request(params = {})
+        spree_put :update, params
+      end
+
+      context 'using guest checkout' do
+        context 'having valid parameters' do
+          before do
+            allow(order).to receive_messages(update_attributes: true)
+            allow(order).to receive_messages(next: false)
+            allow(order).to receive_messages(complete?: true)
+            allow(order).to receive_messages(refresh_shipment_rates: true)
+          end
+
+          context 'having successful response' do
+            before { send_request(attributes) }
+            it { expect(response).to have_http_status(302) }
+            it { expect(response).to redirect_to(edit_admin_order_url(order)) }
+          end
+
+          context 'with correct method flow' do
+            it { expect(order).to receive(:update_attributes).with(attributes[:order]) }
+            it { expect(order).to_not receive(:next) }
+            it { expect(order).to receive(:complete?) }
+            it 'does refresh the shipment rates with all shipping methods' do
+              expect(order).to receive(:refresh_shipment_rates)
+              .with(Spree::ShippingMethod::DISPLAY_ON_FRONT_AND_BACK_END)
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Use 2 (not 0) spaces for indenting an expression spanning multiple lines.